lgn-worker/src/grpc/mod.rs

```diff
@@ -0,0 +1,145 @@
+use anyhow::Result;
+use protobuf::worker_to_gw_request;
+use protobuf::WorkerToGwRequest;
+use protobuf::WorkerToGwResponse;
+use tokio::sync::mpsc;
+use tokio_stream::StreamExt;
+use tonic::metadata::MetadataValue;
+use tonic::Request;
+use tracing::error;
+use tracing::info;
+
+pub mod test;
+
+pub mod protobuf
+{
+    tonic::include_proto!("lagrange");
+}
+const MAX_GRPC_MESSAGE_SIZE_MB: usize = 16;
+
+pub struct GrpcConfig
+{
+    /// Gateway URI
+    pub gateway: String,
+    /// Version of the worker to advertise to the GW
+    pub version: String,
+    /// Token authorization to put in bearer attribute
+    pub token: String,
+    /// worker class to advertise to the Gateway
+    /// TODO: make that a vector to specify multiple classes
+    pub class: String,
+    pub max_grpc_message_size_mb: Option<usize>,
+}
+
+impl GrpcConfig
+{
+    pub fn ready_msg(&self) -> worker_to_gw_request::Request
+    {
```

Comment:
Are those BSD-style braces lgn-coprocessor-specific? Do we really want that everywhere?

Reply:
Found it, that's coming [from here](https://github.com/Lagrange-Labs/lgn-coprocessor/blob/main/rustfmt.toml#L6). 

Don't like it at all either, Rust is typically K&R style, not BSD.